freelist_hmap.go

```diff
@@ -4,12 +4,7 @@ import "sort"
 
 // hashmapFreeCount returns count of free pages(hashmap version)
 func (f *freelist) hashmapFreeCount() int {
-	// use the forwardMap to get the total count
-	count := 0
-	for _, size := range f.forwardMap {
-		count += int(size)
-	}
-	return count
+	return int(f.freePagesCount)
```

Comment:
As such changes can be very subtle, I would feel safer if we 
could test them/verify extensively with logic of this style: 

```
if env.get("BBOLT_VERIFY")='true' {
   if f.freePagesCount != f.slowFreePagesCount() {
     panic(...);
   }
}
```

Even better having a piece of assertion framework: 

```
func assertF(assertFunc func () bool, fmt string, args string) {
  if env.get("BBOLT_VERIFY")='true' {
     if !assertFunc() {
        panic(fmt, args);
     }
 } 
}
```

to write: 

```
  assert(func() { f.freePagesCount == f.slowFreePagesCount()}, "freePagesCount is out of sync with free pages map");
```

Reply:
agree, add an assert to verify it will be safer!
os.GetEnv() will do syscall every single time, so I just init it as a global variable

Reply:
Please follow 
- https://github.com/etcd-io/etcd/blob/main/client/pkg/verify/verify.go

To be clearer, pls add a verify.go under` internal/common`.  All verifications should be disabled in production, and enabled in test by default.

Reply:
@ahrtr, I fear that @xing0821 is no longer motivated to contribute to this PR.  Can you suggest a way to move forward?  I could open a new PR containing their changes, as well as the verify.go change you're suggesting, but I don't want them to lose credit for the work.

Reply:
Thank you @ncabatoff .  I suggest to follow steps roughly below,
- fork a branch on top of this PR; so that @xing0821 is still the author of the first commit.
- rebase your PR;
- add your new change(s) as the new commits.

Does this make sense to you?